src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -83,6 +91,145 @@ public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
     }
 
     /**
+     * From a given genotype, extract a given subset of alleles and return the new SACs
+     * @param g                             genotype to subset
+     * @param allelesToUse                  alleles to subset
+     * @param originalAlleles               the original alleles
+     * @return                              the subsetted SACs
+     */
+    private static int[] subsetSACAlleles(final Genotype g, final List<Allele> allelesToUse, final List<Allele> originalAlleles){
+
+        if ( !g.hasExtendedAttribute(GATKVCFConstants.STRAND_COUNT_BY_SAMPLE_KEY) )
+            return null;
+
+        // we need to determine which of the alternate alleles (and hence the likelihoods) to use and carry forward
+        final int numOriginalAltAlleles = originalAlleles.size() - 1;
+        final int numNewAltAlleles = allelesToUse.size() - 1;
+        final List<Integer> sacIndicesToUse = numOriginalAltAlleles == numNewAltAlleles ? null : determineSACIndicesToUse(originalAlleles, allelesToUse);
+
+        return makeNewSACs(g, sacIndicesToUse);
+    }
+
+    /**
+     * Make a new SAC array from the a subset of the genotype's original SAC
+     *
+     * @param g               the genotype
+     * @param sacIndicesToUse the indices in the SAC to use given the allelesToUse (@see #determineSACIndicesToUse())
+     * @return subset of SACs from the original genotype, the original SACs if sacIndicesToUse is null
+     */
+    private static int[] makeNewSACs(final Genotype g, final List<Integer> sacIndicesToUse) {
+
+        final int[] oldSACs  = getSACs(g);
+
+        if (sacIndicesToUse == null) {
+            return oldSACs;
+        } else {
+            final int[] newSACs = new int[sacIndicesToUse.size()];
+            int newIndex = 0;
+            for (final int oldIndex : sacIndicesToUse) {
+                newSACs[newIndex++] = oldSACs[oldIndex];
+            }
+            return newSACs;
+        }
+    }
+
+    /**
+     * Get the genotype SACs
+     *
+     * @param g the genotype
+     * @return an arrays of SACs
+     * @throws GATKException if the type of the SACs is unexpected
+     */
+    private static int[] getSACs(final Genotype g) {
+
+        if ( !g.hasExtendedAttribute(GATKVCFConstants.STRAND_COUNT_BY_SAMPLE_KEY) )
+            throw new IllegalArgumentException("Genotype must have SAC");
+
+        if ( g.getExtendedAttributes().get(GATKVCFConstants.STRAND_COUNT_BY_SAMPLE_KEY).getClass().equals(String.class) ) {
```

Comment:
It might be slightly more efficient / easier to read if you just get the attribute once and save it in a named variable